test/pack/pack.c

```diff
@@ -60,18 +65,71 @@ typedef enum {
     MEM_TYPE__DEVICE,
 } mem_type_e;
 
-#ifdef HAVE_CUDA
+#if defined(HAVE_CUDA) || defined(HAVE_ZE)
 static int ndevices = -1;
 #endif
 static int device_id = 0;
 static int device_stride = 0;
 
+#ifdef HAVE_ZE
+static ze_driver_handle_t global_ze_driver_handle;
+static ze_device_handle_t *global_devices = NULL;
+static ze_context_handle_t ze_context;
+#endif
```

Comment:
The code is getting very messy.  It might be better to move the CUDA and ZE pieces into separate files, and have `pack.c` simply access them through some interface.

Reply:
done